operators/pkg/instance-controller/common_logic.go

```diff
@@ -0,0 +1,70 @@
+package instance_controller
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	crownlabsv1alpha2 "github.com/netgroup-polito/CrownLabs/operators/api/v1alpha2"
+	instance_creation "github.com/netgroup-polito/CrownLabs/operators/pkg/instance-creation"
+)
+
+// BuildInstanceObjectMeta creates ObjectMeta for Instance.
+func BuildInstanceObjectMeta(
+	name string, namespace string,
+	template *crownlabsv1alpha2.Environment,
+	instanceName string) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      name,
+		Namespace: namespace,
+		Labels: map[string]string{
+			"name":          name,
+			"template-name": template.Name,
+			"instance-name": instanceName,
+		},
+	}
+}
+
+// CreateInstanceExpositionEnvironment creates necessary instance components for accessing the instance (ingress, service and oauth).
+func CreateInstanceExpositionEnvironment(
+	ctx context.Context,
+	r *InstanceReconciler,
+	instance *crownlabsv1alpha2.Instance,
+	name string, namespace string,
```

Comment:
InstanceReconciler should be a receiver and not a parameter:

```suggestion
func (r *InstanceReconciler) CreateInstanceExpositionEnvironment(
	ctx context.Context,
	instance *crownlabsv1alpha2.Instance,
	name string, namespace string,
```

Name and namespace are already available inside the instance IIUC